hyperspy/component.py

```diff
@@ -269,6 +286,15 @@ def _set_free(self, arg):
             self.component._update_free_parameters()
         self.trait_property_changed('free', old_value, self.__free)
 
+    def _on_twin_update(self, value, twin=None):
+        if (twin is None or not hasattr(twin, 'events') or
+                not hasattr(twin.events, 'value_changed')):
+            self.events.value_changed.trigger(value=value, parameter=self)
+        else:
+            with twin.events.value_changed.suppress_callback(
+                    self._on_twin_update):
+                self.events.value_changed.trigger(value=value, parameter=self)
+
```

Comment:
I guess it is probably I who wrote this, but I would invert the logic and switch the order of the if/else brackets:
```python
if (twin is not None and hasattr(twin, 'events') and hasattr(twin.events, 'value_changed')):
```